test/sql/join/test_not_distinct_from.test

```diff
@@ -0,0 +1,124 @@
+# name: test/sql/join/test_not_distinct_from.test
+# description: Test join on is not distinct from query
+# group: [join]
+
+
+statement ok
+create table tbl_1 (a integer, b integer)
```

Comment:
Extra tests: 
* More types (all numeric types, strings, dates, times, timestamps, interval)
* Use on a bigger table (~10K elements)
* Use with filter (e.g. `SELECT x IS NOT DISTINCT FROM y FROM tbl WHERE z%2=0`, where the expression filters like 50% of the rows)
* Use in DEFAULT expression of a table
* Use in DEFAULT expression of a table and store that table on disk (storage test)
* Use with rowid column (non-flat and non-constant vector)
* Use in subqueries with correlated expressions